### DISCUSSION TITLE
Unified Input types

### DISCUSSION THREAD

**Discussion by matchatype** [tailwindlabs/tailwindcss#1798] [Ideas]
I really miss more advanced input types, like date/time/range/calendar, file, etc... all of which do have default rendering settings and you most likely want to style consistently with the rest of the app. These elements should not be left out of tailwind.

**Comment by estevanmaito** [comment]
Are you talking about Tailwind CSS or the product Tailwind UI?

Because Tailwind CSS doesn't include any input at all. You have to add every style even for a simple text input.

**Comment by matchatype** [comment]
I'm talking about Tailwind CSS. I do understand the utility-first approach, however I miss examples for all those not so straightforward situations where the "for rapidly building custom designs" statement loses some of its original shine. It's not so uncommon the need to style these elements, and I think examples should be covered among the forms section of Tailwind CSS website.

It's just an idea I'd like to discuss, and the reason why I didn't open an issue instead (then you'd be right, as the framework does not include any element-specific style, only tools to provide for them).

**Reply by mgibas** [reply]
TailwindUI seems to fit your requirements - worth every penny!

**Reply by matchatype** [reply]
I agree, that seems like an alternative appropriate place to ship such preconfigured components, however I don't see any picker, neither among published components nor on the [public roadmap](https://trello.com/b/Nh6e9w8Q/tailwind-ui-roadmap). So actually that doesn't fit the requirements.

**Reply by mgibas** [reply]
Yeah I believe it’s still in early stage.

**Reply by matchatype** [reply]
Maybe, but apparently there's no mention even on the roadmap, so I doubt pickers will see the light over there.

**Reply by adamwathan** [reply]
The roadmap isn't exhaustive, we'll definitely be adding datepickers, file inputs, etc. 👍
